src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -896,10 +909,13 @@ protected virtual void ValidateConditions(Saml2SecurityToken samlToken, TokenVal
             if (samlToken.Assertion == null)
                 throw LogArgumentNullException(nameof(samlToken.Assertion));
 
+            if (validationParameters.RequireAudience && samlToken.Assertion.Conditions == null)
```

Comment:
>samlToken.Assertion.Conditions == null [](start = 56, length = 38)

for saml tokens you are also looking for conditions.count == 0, should the same be done here? (will be caught later, however for consistency)?

Reply:
Its related to we check lifetime before audience.

Reply:
@brentschmaltz If `Conditions.Count() == 0` we'll end up returning early in the case of SamlSecurityTokenHandler, so we'll never call ValidateLifetime. However, in the case of Saml2SecurityTokenHandler we will. Shouldn't we be consistent about this?

Reply:
I got rid of the Conditions.Count() == 0 in the SamlSecurityTokenHandler so that we're consistent.  Even if Conditions.Count() == 0 it's possible we might still need to call ValidateLifetime (since NotBefore and NotOnOrAfter are attributes on the Condition).